### ISSUE TITLE
install --no-dev still requires extensions in require-dev

### ISSUE THREAD

**Issue by nipil** [composer/composer#6416]
Remarks : GD is just an example, i get the same behavior with any extension **not** available

My `composer.json` 

```json
{
    "require-dev": {
        "ext-gd": "*"
    }
}
```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
php composer.phar install --no-dev
```

I get the following output:

```
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.
  Problem 1
    - The requested PHP extension ext-gd * is missing from your system. Install or enable PHP's gd extension.
```

And I expected this to happen:

I want it to install without problem, as i specifically ask not to install developper stuff.

Logically, extensions in required-dev should *not* be required when installing with --no-dev

Anyway, that is how i see it ;-)

**Comment by alcohol** [comment]
Please search existing issues before reporting something that has already been reported.

The behaviour is as expected.

See #5224, #5227, #5444, #5547, #5523

**Comment by nipil** [comment]
Thanks for pointing out these other issues, especially 5444.

I agree this is not a *bug*, as the discussions in these issues say it's the expected behaviour... ok, i can accept that.

But this behaviour is really illogical and counterintuitive, that's what i find bothering.

I openend another issue, to try to change the behaviour so that it makes sense !

**Comment by alcohol** [comment]
Please do not open another issue... it has already been discussed/debated before as you can see. The current implementation is not going to be changed.

**Comment by nipil** [comment]
I was merely explaining why i opened "another" (ie this current request, in regards to the other closed ones you mentionned in your first reply) : i'm not going to open a new one, as it would be pointless
